enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,264 @@
+
+As mentioned in the above section, degrading will only happen in two scenarios:
+- Translating the ignition stub to spec 3 fails. This is likely more fatal and won't get fixed without the editing the ignition stub manually.
+- Patching of the MachineSet fails. This is likely due to a temporary API server outage and will resolve itself without user intervention.
+
+Every degrade will be associated with a machineset. As a result, the MSC will have to maintain a local list of currently "degraded" machinesets, and remove/add to this list during a sync loop as necessary. Based on the changes to this list, the MSC can then intiate, update or clear a degrade condition on the operator.
+
+#### Reverting to original bootimage
+
+Few strategies here:
+
+##### via Annotations
+
+The MSC will maintain a backup of the the following via annotation to the `MachineSet`:
+- `io.openshift.mco-managed-factory-image=` storing the original provider image reference prior to when the feature was turned on
+- `io.openshift.mco-managed-factory-secret=` storing the original stub secret name
+- `io.openshift.mco-managed-last-image=` storing the last provider image reference before an update
+- `io.openshift.mco-managed-last-secret=` storing the last stub secret name before an update
+
+##### via ConfigMap
+
+The MSC will maintain a backup of the following per machinset, with these keys:
+- `$(machine_set_name)-factory-image` storing the original provider image reference prior to when the feature was turned on
+- `$(machine_set_name)-factory-secret` storing the original stub secret name prior to the feature was turned on
+- `$(machine_set_name)-last-image` storing the last provider image reference before an update
+- `$(machine_set_name)-last-secret` storing the last stub secret name before an update
+
+##### Via new MachineSet fields
+
+Another proposed strategy is to add the required fields into the `.Status` section of a `MachineSet` object. This would be an API change.
+
+##### Mechanism
+
+A revert can be done by opting out the `MachineSet`, this will trigger the MSC to restore the MachineSet to last known good values stored from the backups. This is an important mitigation in case things go wrong(invalid bootimage references, incorrect patching... etc).
+
+The reason for keeping a factory version is in case the admin wishes to restore to factory values manually(they will have to opt-out of the feature first). It may also may aid in debugging.
+
+### Workflow Description
+
+It is important to note that there would be two "opt-in" knobs while this feature is under TechPreview. The admin would first have to turn on the feature gate, and then the opt-in mechanism. The secondary knob is necessary as some customers may want to keep their boot images static when this feature leaves TechPreview.
+Couple of strategies as before:
+
+##### via Annotations
```

Comment:
Can we drop this (/move to alternatives), an annotation is not an API and not a suitable way to opt this feature in. We should use the operator type approach only

Reply:
Will drop this!